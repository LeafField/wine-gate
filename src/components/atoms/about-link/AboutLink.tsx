import React, { FC } from "react";
import Link from "next/link";
import Image from "next/image";
import AboutImage from "../../../images/about_link_image.jpg";

const AboutLink: FC = () => {
  return (
    <Link
      className="group relative row-span-1 block w-full overflow-hidden tablet:col-span-2 laptop:col-span-1"
      href={"/about"}
    >
      <figure className="absolute h-full w-full transition-transform duration-300 group-hover:scale-110">
        <Image
          src={AboutImage}
          alt="棚に並ぶワインボトル"
          fill
          style={{ objectFit: "cover" }}
          priority
          sizes="(min-width:768px) 332px, 100vw"
        />
      </figure>
      <div className="relative z-front flex h-full w-full bg-image-cover align-bottom">
        <h2 className="mt-auto w-full px-6 py-4 text-end text-heading3 text-white">
          このサイトについて
        </h2>
      </div>
    </Link>
  );
};

export default AboutLink;
