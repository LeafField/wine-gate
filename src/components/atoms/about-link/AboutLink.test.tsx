import { screen, render } from "@testing-library/react";
import AboutLink from "./AboutLink";

describe("AboutLinkの単体テスト", () => {
  test("スナップショットテスト", () => {
    const { container } = render(<AboutLink />);
    expect(container).toMatchSnapshot();
  });
});
