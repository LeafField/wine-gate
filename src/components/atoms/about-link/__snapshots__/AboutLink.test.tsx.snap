// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutLinkの単体テスト スナップショットテスト 1`] = `
<div>
  <a
    class="group relative row-span-1 block w-full overflow-hidden tablet:col-span-2 laptop:col-span-1 "
    href="/about"
  >
    <figure
      class="absolute h-full w-full transition-transform duration-300 group-hover:scale-110"
    >
      <img
        alt="棚に並ぶワインボトル"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </figure>
    <div
      class="relative z-front flex h-full w-full bg-image-cover align-bottom"
    >
      <h2
        class="mt-auto w-full px-6 py-4 text-end text-heading3 text-white"
      >
        このサイトについて
      </h2>
    </div>
  </a>
</div>
`;
